packages/php-db-import-export/src/Backend/Bigquery/BigqueryException.php

```diff
@@ -25,18 +27,81 @@ public static function covertException(JobException|ServiceException $e): Throwa
     public static function createExceptionFromJobResult(array $jobInfo): Throwable
     {
         $errorMessage = $jobInfo['status']['errorResult']['message'] ?? 'Unknown error';
+        $jobErrors = $jobInfo['status']['errors'] ?? [];
 
         // detecting missing required column. Record with `Required column value is missing` substring contains
         // much better information for enduser
-        foreach ($jobInfo['status']['errors'] ?? [] as $error) {
+        foreach ($jobErrors as $error) {
             if (str_contains($error['message'], 'Required column value is missing')) {
                 $errorMessage = $error['message'];
                 return new BigqueryInputDataException($errorMessage);
             }
         }
 
+        $filteredJobErrors = array_filter($jobErrors, function ($error) use ($jobInfo) {
+            // the errorResult is the first in list of errors as well
+            return $error['message'] !== $jobInfo['status']['errorResult']['message'];
+        });
+        $countOfErrors = count($filteredJobErrors);
+        $parsingErrors = array_filter($filteredJobErrors, function ($error) {
+            return self::isUserError($error['message'], $error['reason']);
+        });
+        if (count($parsingErrors) > 0) {
+            // filter parsing errors
+            $areExtraErrors = count($parsingErrors) !== $countOfErrors;
+            return new BigqueryInputDataException(self::getErrorMessageForErrorList($parsingErrors, $areExtraErrors, $jobInfo['jobReference']['jobId']));
+
+            if (count($filteredJobErrors) > 0 && count($filteredJobErrors) < 10) {
+                // if there is reasonable number of errors, we can output them all
+                $errors = implode(PHP_EOL, array_map(function ($error) {
+                    return $error['message'];
+                }, $filteredJobErrors));
+                return new BigqueryInputDataException('CSV processing failed:' . PHP_EOL . $errors);
```

Comment:
proc `CSV processing`? prvne - jednou muzeme klidne prejit na parquet, druhak - tady ten handler se muze pouzit na hromade mist

Reply:
Fair! Ono je to v té původní erorr message: `Error while reading data, error message: CSV processing encountered too many errors, giving up. `

Reply:
@tomasfejfar tomu nerozumim proc jsi celou tu logiku zpracovani erroru [dropnul](https://github.com/keboola/storage-backend/pull/120/commits/117decc4410653266dbe143e8fbcf0d02d9680a0). ta je imho validni, mam jen problem s tim zminenim `CSV` v nasi hlasce ?

Reply:
Protože se nevolala. byl to dead code.

Reply:
https://github.com/keboola/storage-backend/pull/120/commits/117decc4410653266dbe143e8fbcf0d02d9680a0#diff-6e795f8d0f1872899677f6e5c4dcea1af5e3ca0d9f5bea1d9a0748e3ece28523R52

Reply:
jasne, omg..